src/FsAutoComplete/CodeFixes/AdjustConstant.fs

```diff
@@ -0,0 +1,1517 @@
+/// * returns contained constant when `pos` inside contained constant
+/// * otherwise `SynConst.Measure` when on other parts of `Measure` constant (`<km>`)
+///
+/// Note:  
+/// Might be erroneous Constant -> containing `value` is then default (`0`).
+/// Check by comparing returned range with existing Diagnostics.
+let private tryFindConstant ast pos =
+  let rec findConst range constant =
+    match constant with
+    | SynConst.Measure (constant, constantRange, _) when rangeContainsPos constantRange pos ->
+        findConst constantRange constant
+    | _ -> (range, constant)
+  SyntaxTraversal.Traverse(
+    pos,
+    ast,
+    { new SyntaxVisitorBase<_>() with
+        member _.VisitExpr(_, _, defaultTraverse, expr) =
+          match expr with
+          //                                     without: matches when `pos` in comment after constant
+          | SynExpr.Const (constant, range) when rangeContainsPos range pos ->
+              findConst range constant
+              |> Some
+          | _ -> defaultTraverse expr
+        member _.VisitEnumDefn(_, cases, _) =
+          cases
+          |> List.tryPick (fun (SynEnumCase(valueExpr = expr)) ->
+              let rec tryFindConst (expr: SynExpr) =
+                match expr with
+                | SynExpr.Const (constant, range) when rangeContainsPos range pos ->
+                    findConst range constant
+                    |> Some
+                | SynExpr.Paren (expr=expr) -> tryFindConst expr
+                | SynExpr.App (funcExpr=funcExpr) when rangeContainsPos funcExpr.Range pos ->
+                    tryFindConst funcExpr
+                | SynExpr.App (argExpr=argExpr) when rangeContainsPos argExpr.Range pos ->
+                    tryFindConst argExpr
+                | _ -> None
+              tryFindConst expr
+          )
+        member _.VisitPat(_, defaultTraverse, synPat) =
+          match synPat with
+          | SynPat.Const (constant, range) when rangeContainsPos range pos ->
+              findConst range constant
+              |> Some
+          | _ -> defaultTraverse synPat
+    }
+  )
+
+/// Computes the absolute of `n`
+///
+/// Unlike `abs` or `Math.Abs` this here handles `MinValue` and does not throw `OverflowException`.
+type private Int =
+  static member inline abs(n: sbyte): byte =
+    if n >= 0y then byte n else byte (0y - n)
+  static member inline abs(n: int16): uint16 =
+    if n >= 0s then uint16 n else uint16 (0s - n)
+  static member inline abs(n: int32): uint32 =
+    if n >= 0l then uint32 n else uint32 (0l - n)
+  static member inline abs(n: int64): uint64 =
+    if n >= 0L then
+      uint64 n
+    else
+      // unchecked subtraction -> wrapping/modular negation
+      // -> Negates all numbers -- with the exception of `Int64.MinValue`:
+      //        `0L - Int64.MinValue = Int64.MinValue`
+      //        BUT: converting `Int64.MinValue` to `UInt64` produces correct absolute of `Int64.MinValue`
+      uint64 (0L - n)
```

Comment:
Is there a way to get a generic `abs': 'int -> 'uint`? (`abs'` just calls corresponding `Int.abs`)   
Extension Methods cannot (yet) be used with SRTPs. So I need to redirect to my `Int` type. But then I couldn't manage to correctly resolve `abs'` without additional hints 

Best I managed was:
```fsharp
let inline abs'<'Int, 'int, 'uint when 'Int :> Int and 'Int:(static member abs: 'int -> 'uint)> (value: 'int) : 'uint =
  'Int.abs value
```
But `abs' -5` does not work. I have to either give a return type hint or specify generic types (as wildcard...)
```fsharp
// what I want:
printfn "|%i| = %i" -5 (abs' -5)
//                      ^^^^
//                      error FS0071: Type constraint mismatch when applying the default type 'Int' for a type inference variable. 
//                            This expression was expected to have type 'int' but here has type 'uint32'

// what works:
printfn "|%i| = %i" -5 (abs' -5 : uint)
printfn "|%i| = %i" -5 (abs'<_,_,_> -5)
```

Not something I use here in this PR (or usually somewhere else). But I used this PR to experiment a bit (too much...) with `Span`, inline and Generic Numbers. And `abs'` is one of the cases that came up, but couldn't get the result I wanted :(


(Another issue was with `MaxValue` & `MinValue`: Seems like SRTP doesn't support Constants/Fields?)

Reply:
Gotta look to F#+ for all the real hard work.  I hate that I know how to do this.

```fsharp

type Default1 = class end
type Int =
  inherit Default1

  static member inline AbsU(n: sbyte,_method : Int) : byte = if n >= 0y then byte n else byte (0y - n)
  static member inline AbsU(n: int16,_method : Int) : uint16 = if n >= 0s then uint16 n else uint16 (0s - n)
  static member inline AbsU(n: int32,_method : Int) : uint32 = if n >= 0l then uint32 n else uint32 (0l - n)
  static member inline AbsU(n: int64,_method : Int) : uint64 =
    if n >= 0L then
      uint64 n
    else
      // unchecked subtraction -> wrapping/modular negation
      // -> Negates all numbers -- with the exception of `Int64.MinValue`:
      //        `0L - Int64.MinValue = Int64.MinValue`
      //        BUT: converting `Int64.MinValue` to `UInt64` produces correct absolute of `Int64.MinValue`
      uint64 (0L - n)

  static member inline AbsU(n: nativeint,_method : Int) : unativeint = if n >= 0n then unativeint n else unativeint (0n - n)

  static member inline Invoke (x)  : '``Abs<T>`` =
      let inline call (mthd: ^M, source: ^I, _output: ^R) = ((^M or ^I or ^R) : (static member AbsU : _*_ -> _) x, mthd)
      call (Unchecked.defaultof<Int>, x, Unchecked.defaultof<'``Abs<'U>``>)


let inline absU x = Int.Invoke x

let doIt () =

  let byteEx = (absU -5y)
  let uint16Ex = (absU -5s)
  let uint32Ex = (absU -5l)
  let uint32Ex = (absU -5l)
  let uint64Ex = (absU -5L)
  let unativeIntEx = (absU -5n)

  ()
```

Reply:
Ok ... I think I stick with passing the concrete `abs`/`AbsU` around .... 


But always interesting to see what's possible with F#. Thanks :thumbsup: